src/enumValues.ts

```diff
@@ -4,35 +4,22 @@ export class EnumValues {
   }
 
   static getNames(e: any) {
-    if (this.isNumericEnum(e)){
-      return this.getObjectValues(e).filter(v => typeof v === "string") as string[];
-    }
+    const skinnyE = this.removeNumberKeys(e);
 
-    return Object.keys(e);
+    return Object.keys(skinnyE);
   }
 
-  static getValues(e: any) {
-    return this.getNames(e).map(name => e[name]) as number[];
+  private static removeNumberKeys(e: any) {
```

Comment:
Good eye for the pattern! But you can do it much simpler. You can delete this method and simply use:
```return Object.keys(e).filter(key => isNaN(key))``` in the implementation of the getNames